Build fails

```
In file included from primme/src/svds/primme_svds_interfacedouble.c:3:
In file included from primme/src/svds/primme_svds_interface.c:41:
In file included from primme/src/include/numerical.h:36:
primme/src/include/template.h:106:4: error: "An arithmetic should be selected, please define one of USE_DOUBLE, USE_DOUBLECOMPLEX, USE_FLOAT or USE_FLOATCOMPLEX."
#  error "An arithmetic should be selected, please define one of USE_DOUBLE, USE_DOUBLECOMPLEX, USE_FLOAT or USE_FLOATCOMPLEX."
   ^
In file included from primme/src/svds/primme_svds_interfacedouble.c:3:
In file included from primme/src/svds/primme_svds_interface.c:41:
In file included from primme/src/include/numerical.h:36:
In file included from primme/src/include/template.h:372:
/usr/include/malloc.h:3:2: error: "<malloc.h> has been replaced by <stdlib.h>"
#error "<malloc.h> has been replaced by <stdlib.h>"
 ^
```


Please note that the ```<malloc.h>``` header has been discontinued, see here for Linux: https://linux.die.net/man/3/malloc

OS: FreeBSD 11.2

--- a/src/eigs/primme_interface.c
+++ b/src/eigs/primme_interface.c
@@ -36,7 +36,7 @@
  *
  ******************************************************************************/
 
-#if !(defined (__APPLE__) && defined (__MACH__))
+#if !((defined (__APPLE__) && defined (__MACH__)) || (defined (__FreeBSD__)))
 #  include <malloc.h>
 #endif
 #include <stdlib.h>   /* mallocs, free */

--- a/src/include/template.h
+++ b/src/include/template.h
@@ -376,7 +376,7 @@
  *
  **********************************************************************/
 
-#if !(defined (__APPLE__) && defined (__MACH__))
+#if !((defined (__APPLE__) && defined (__MACH__)) || (defined (__FreeBSD__)))
 #  include <malloc.h> /* malloc */
 #endif
 #include <stdlib.h>   /* malloc, free */

